### PR TITLE
Lessons

### PR DESCRIPTION
**Pull request by Khaledgarbaya**
This PR will implement the lessons section of the tutorial

### CODE REVIEW

File: app/routes/tutorials/$tutorialSlug/lessons/index.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+}
+export default function Lessons() {
+  const { lesson } = useLoaderData<typeof loader>()
+  return (
+    <main className='relative mx-auto w-full max-w-7xl items-start border-t border-transparent lg:mt-16 2xl:flex 2xl:border-gray-800'>
+      <div className='flex flex-col border-gray-800 2xl:relative 2xl:h-full 2xl:w-full 2xl:border-r'>
+        <div className='relative flex w-full items-center justify-center'>
+          <iframe
+            className='w-full aspect-video'
+            src={`${lesson?.fields?.videoUrl}`}
```
Comment (machour) [inline app/routes/tutorials/$tutorialSlug/lessons/index.tsx]: by now, we should now that `lesson` is not undefined, right?

**Review by machour — COMMENTED** [review]
Kind of hard to review the changes with the double quote => single quote changes.
Ken tnejjem taamel rollback, w mba3d fi PR okhra treformati, sinon, bagla liha 🤷🏼

File: app/routes/tutorials/$tutorialSlug/lessons/index.tsx (hunk 1 above)
Comment (Khaledgarbaya) [reply app/routes/tutorials/$tutorialSlug/lessons/index.tsx]: old react habits lol

**Comment by Khaledgarbaya** [comment]
ok fixed the quote diff

File: app/root.tsx [hunk 2]
```diff
@@ -84,41 +84,87 @@ function Document({
...
+          <nav className="container mx-auto flex justify-between items-center">
+            <ul className="flex">
+              <li className="p-4">
+                <NavLink
+                  className={({ isActive }) =>
+                    `${
+                      isActive
+                        ? 'border-b border-b-amber-500 text-amber-600'
+                        : 'text-white'
+                    }`
```
Comment (machour) [inline app/root.tsx]: Not a `clsx` fan?

File: app/routes/tutorials/$tutorialSlug/index.tsx [hunk 3]
```diff
@@ -35,13 +35,21 @@ export default function Index() {
...
           }}
         />
+        {tutorial.fields.lessons && tutorial.fields.lessons.length > 0 ? (
+          <Link
+            to={`lessons/${tutorial.fields.lessons[0].fields.slug}`}
+            className="group relative flex w-full justify-center rounded-md border border-transparent bg-amber-600 py-2 px-4 text-sm font-medium text-white hover:bg-amber-700 focus:outline-none focus:ring-2 focus:ring-amber-500 focus:ring-offset-2"
+          >
+            Start Learning
+          </Link>
+        ) : null}
```
Comment (machour) [inline app/routes/tutorials/$tutorialSlug/index.tsx:58]: You're returning all the tutorial object from your loader, just to read its title, content and slug.
How many more fields are present in the tutorial ?

Depending on your answer, you could return just the `title`, `content` and `slug` variables from your loader, and keep a JSX clean (no ifs)

**Review by machour — COMMENTED** [review]
some remarks, still didn't go through everything

File: app/root.tsx (hunk 2 above)
Comment (Khaledgarbaya) [reply app/root.tsx]: I though this is a simple use case as these are the only styles to apply what I will be doing here is replacing the backtick with `clsx`

**Comment by machour** [comment]
You have renicated quotes :D

**Comment by Khaledgarbaya** [comment]
Merging this as I think we are 90% there, next will be optimization

**Pull request merged**